q2_qemistree/_plot.py

```diff
@@ -96,10 +96,51 @@ def format_labels(feature_metadata, category, ms2_label, parent_mz):
     return '\n'.join(labels)
 
 
-def plot(output_dir: str, table: biom.Table, tree: NewickFormat,
-         feature_metadata: pd.DataFrame, category: str,
-         color_palette: str = 'Dark2', ms2_label: bool = False,
-         parent_mz: str = None) -> None:
+def format_barplots(table: biom.Table):
+    barplots = []
+    barplots.append('DATASET_MULTIBAR')
+    barplots.append('SEPARATOR TAB')
+    barplots.append('DATASET_LABEL\tRelative Abundance')
+    barplots.append('COLOR\t#ff0000')
+
+    table = table.norm(axis='observation', inplace=False)
+    table = pd.DataFrame(table.matrix_data.toarray(), table.ids('observation'),
```

Comment:
I guess you can do `table = table.to_dataframe()`

Reply:
`to_dataframe()` is slow in my experience

Reply:
`to_dataframe` is slow if you use a sparse dataframe, if you force it to a dense dataframe it is much faster. To change this use `table.to_dataframe(True)`. Alternatively, I don't see anything in the code that makes this conversion necessary, all the data you are using can be accessed from the Table object. For example I think the loop in line 123 can be done using `Table.iter_data`:

http://biom-format.org/documentation/generated/biom.table.Table.iter_data.html#biom.table.Table.iter_data